### ISSUE TITLE
"commandline -r [CMD]" does not place CMD on the command line buffer

### ISSUE THREAD

**Issue by imcaizheng** [fish-shell/fish-shell#9120]
``` fish
> fish --verson
fish, version 3.4.1

> uname -a
Linux caizheng-nuc10i7fnk 5.10.131-1-MANJARO #1 SMP PREEMPT Fri Jul 15 09:45:13 UTC 2022 x86_64 GNU/Linux
```
## Problem
Upgraded from version 3.3.1, fish shell behaves different around the `commandline` command. For example, `commandline -r "/bin/"(ls /bin | fzf)` used to work as expected and would change the current command line buffer but with version 3.4.1 the command outputs nothing and exits silently.

What I expected is that commands like `commandline -r`, `commandline -a` and `commandline -i` can actually change current command line buffer once being executed by a user.

## Other Info
Seems key bindings still work well with the `commandline` command.

Tried creating a key binding with `bind \cq 'commandline -r "/bin/"(ls /bin | fzf)'` and after pressing Ctrl-q, finishing fzf selection, it updated the command line buffer with the selected item.

**Comment by krobelus** [comment]
Duplicate of #8807

**Comment by krobelus** [comment]
fixed in 3.5.0
